backend/fms_core/services/project_link_samples.py

```diff
@@ -0,0 +1,44 @@
+from django.core.exceptions import ValidationError
+from fms_core.models import SampleByProject
+
+def create_link(sample=None, project=None):
+    project_sample_link = None
+    errors = []
+    warnings = []
+
+    if not all([sample, project]):
+        errors.append(f"Unable to process sample or project information.")
+        return (project_sample_link, errors, warnings)
+
+    if SampleByProject.objects.filter(sample=sample, project=project).exists():
+        errors.append(f"[Sample {sample.name}] is already associated to project [{project.name}].")
+        return (project_sample_link, errors, warnings)
+
+    link_data = dict(sample=sample, project=project)
+
+    try:
+        project_sample_link = SampleByProject.objects.create(**link_data)
```

Comment:
Very minor detail, but why not plugging your dictionary directly it is not re-used so no need to make a variable link_data out of it. ex.: deleting line 17, and in line 20 having in the part SampleByProject.objects.create(sample=sample, project=project)